### ISSUE TITLE
Uninitialized Unknown Column Warning

### ISSUE THREAD

**Issue by PinkAppleFlower** [rstudio/rstudio#9138]
IMPORTANT: This bug cannot reliably be reproduced, but I was still asked to file it.

### System details

    RStudio Edition : Desktop
    RStudio Version : Version 1.4.1106: "Tiger Daylily" (2389bc24, 2021-02-11) for Windows
    OS Version      : Windows 10 Pro for Workstations
    R Version       : 4.0.4 -- "Lost Library Book"

### Steps to reproduce the problem
It is not clear to me what steps are exactly needed to reproduce the problem.  I did have the tibble library loaded.  Please see the attached file for example code.

### Describe the problem in detail
When running the code, a warning stating that there is an unknown or uninitialized column will appear in the output. The error might be in the diagnostics system, which is erroneously triggering tibble code when trying to run.  Closing and restarting RStudio did not solve the error for me when it was occurring consistently.  Some people mentioned that the warning would even appear for columns not in the currently running script.

### Describe the behavior you expected
No warning should be triggered by the code when executing.  The error did not occur when ran in the RGUI ever.  

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).

- [ ] I have installed the latest version of RStudio, and confirmed that the issue still persists.
 The warning is not consistently reproducible. So the above, is not really very applicable, but I do have the most recent version installed and the error occurred in the RStudio version prior to the most recent version for me, I believe in Windows 10 Pro, not a workstation version.

- [n.a.] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
 The bug does not cause RStudio to crash.

- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
Attached is a subset of the larger file that triggered the warning.  This file, when ran independently of the large file did not consistently trigger the error, but did when ran after the larger file.  The warning was triggered when running the ROC function code and stated that columns L and f were uninitialized. This example is relatively verbose, but is the only one I have.

[PersistantUnknown.zip](https://github.com/rstudio/rstudio/files/6185762/PersistantUnknown.zip)

**Comment by ronblum** [comment]
@PinkAppleFlower Thank you for raising this! I can't reproduce the error, though as you point out, it's not consistent.

A couple questions:
1) Does this happen when you run chunks individually or knit the file?
2) Can you paste the exact warning that you get?
3) When the warning occurs, which chunk results in the warning? I'm not sure if it's the one that begins with
```
#' ROC curve code
```
or
```
ROC <- get_roc(L = df$y, f = df$predictions)
```

Thanks!

**Comment by PinkAppleFlower** [comment]
@ronblum Hey, no problem. Thanks for looking at it.

1. It happened when I ran the chunk individually, but since it is an R notebook, every time I would save it, the notebook knitted to html output automatically too...  I did not check this output for the error when it was actually happening. I just saw it in the console output when I ran the chunk.


2.  I am not able to reproduce the error on my most recent machine with the most recent version of RStudio.  The error has been discussed on the RStudio Community website and the posts at this link show the exact error message:  https://community.rstudio.com/t/persistent-unknown-or-uninitialised-column-warnings/64879 .  

Kevin Ushey has been following the issue.

3. The chunk that yielded the error is the first one that you mention, #'ROC curve code.

Cheers
